### PR TITLE
dbr quant: stop overridding tensor getters

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70200 dbr quant: start recording ops which are not quantizeable
* #70196 dbr quant: start reusing static quant module mappings
* #70128 dbr quant: better fix for bug with recursion on dequantize
* **#70115 dbr quant: stop overridding tensor getters**
* #70114 dbr quant: clean up logging format
* #70109 dbr quant: add test for qconfig_dict and methods

Summary:

This PR turns off DBR quant __torch_function__ overrides on
tensor attribute getters such as `x.dtype`. This should help
with making the debug logs more readable, and reduce framework
overhead.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D33189544](https://our.internmc.facebook.com/intern/diff/D33189544)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33189544).

File: torch/ao/quantization/_dbr/auto_trace.py [hunk 1]
```diff
@@ -85,11 +85,14 @@ class QuantizationPrepareTensorProxy(torch.Tensor):
         @classmethod
         def __torch_function__(cls, func, types, args=(), kwargs=None):
             nonlocal global_disable_torch_function_override
-            if global_disable_torch_function_override:
-                return super().__torch_function__(func, types, args, kwargs)
-
-            if func == torch.Tensor.__repr__:
+            if (
```
Comment (jerryzh168) [inline torch/ao/quantization/_dbr/auto_trace.py:88]: should we define a helper function

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/quantization/_dbr/auto_trace.py (hunk 1 above)
Comment (vkuzo) [reply torch/ao/quantization/_dbr/auto_trace.py:88]: maybe if n gets from 2 to 3?  going to keep it for now but will keep an eye for standardizing this later, hope that's ok?

Comment (jerryzh168) [reply torch/ao/quantization/_dbr/auto_trace.py:88]: sure, sounds good

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33189544).

**Pull request closed without merging**